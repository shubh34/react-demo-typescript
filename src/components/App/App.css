.App {
  background-color: #eeeeee;
  color: #4183c4;
  height: 500px;
  margin: 0 auto;
  box-shadow: 0px 5px 5px 5px #EEE
}
@media only screen and (min-width: 1280px) {
  .App {
    max-width: 1280px
  }
}

