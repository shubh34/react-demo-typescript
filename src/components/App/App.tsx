import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router, Route, Link, Switch } from "react-router-dom";
import 'semantic-ui-css/semantic.min.css';
import './App.css';
import store from '../../store';
import BlogsListing from '../BlogsListing/BlogsListing';
import PageHeader from '../PageHeader/PageHeader';
import CreateBlog from '../CreateBlog/CreateBlog';
import ViewBlog from '../ViewBlog/ViewBlog';

const App: React.FC = () => {
  return (
    <Provider store={store()}>
      <Router>
        <div className="App">
          <PageHeader />
          <Switch>
            <Route exact path="/" component={BlogsListing} />
            <Route exact path="/addBlog" component={CreateBlog} />
            <Route exact path="/viewBlog/:id" component={ViewBlog} />
            <Route component={BlogsListing} />
          </Switch>
        </div>
      </Router>
    </Provider>
  );
}

export default App;
