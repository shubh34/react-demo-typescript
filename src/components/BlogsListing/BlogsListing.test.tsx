import React from 'react';
import { shallow } from 'enzyme'
import App from './BlogsListing';

it('renders without crashing', () => {
  const div = shallow(<App />)
  expect(div).toMatchSnapshot()
});
