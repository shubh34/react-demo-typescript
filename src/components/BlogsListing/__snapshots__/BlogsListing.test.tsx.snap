// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
Object {
  "props": Object {
    "addBlog": [MockFunction],
    "blogs": Array [
      Object {
        "categories": Array [
          "Food",
          "Travel",
        ],
        "content": "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do.",
        "date": "21 Jan 2019",
        "id": "1",
        "readMore": true,
        "title": "Food Blog",
      },
      Object {
        "categories": Array [
          "Dance",
          "Learning",
        ],
        "content": "Lorem ipsumutpat consequat mus.",
        "date": "22 Jan 2019",
        "id": "2",
        "readMore": false,
        "title": "Dance Blog",
      },
    ],
    "deleteBlog": [MockFunction],
  },
  "wrapper": <div>
    Blog Listing
    <Blog
      blog={
        Object {
          "categories": Array [
            "Food",
            "Travel",
          ],
          "content": "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do.",
          "date": "21 Jan 2019",
          "id": "1",
          "readMore": true,
          "title": "Food Blog",
        }
      }
      deleteBlog={[MockFunction]}
      key="1"
    />
    <Blog
      blog={
        Object {
          "categories": Array [
            "Dance",
            "Learning",
          ],
          "content": "Lorem ipsumutpat consequat mus.",
          "date": "22 Jan 2019",
          "id": "2",
          "readMore": false,
          "title": "Dance Blog",
        }
      }
      deleteBlog={[MockFunction]}
      key="2"
    />
    <Button
      as="button"
      onClick={[Function]}
    >
      Add Blog
    </Button>
  </div>,
}
`;
